pages/spacedeck/[[...roomId]].js

```diff
@@ -0,0 +1,309 @@
+import getConfig from 'next/config';
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { isEmpty } from 'lodash';
+import _ from 'lodash';
+import { useRouter } from 'next/router';
+
+import LoadingSpinner from '../../components/UI/LoadingSpinner';
+import LoadingSpinnerInline from '../../components/UI/LoadingSpinnerInline';
+import { useAuth } from '../../lib/Auth';
+import { useMatrix } from '../../lib/Matrix';
+import ErrorMessage from '../../components/UI/ErrorMessage';
+import Bin from '../../assets/icons/bin.svg';
+import { ServiceSubmenu } from '../../components/UI/ServiceSubmenu';
+import IframeLayout from '../../components/layouts/iframe';
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import Form from '../../components/UI/Form';
+import CopyToClipboard from '../../components/UI/CopyToClipboard';
+import ServiceLink from '../../components/UI/ServiceLink';
+
+export default function Spacedeck() {
+    const auth = useAuth();
+    const matrix = useMatrix(auth.getAuthenticationProvider('matrix'));
+    const matrixClient = auth.getAuthenticationProvider('matrix').getMatrixClient();
+    const { t } = useTranslation('spacedeck');
+    const router = useRouter();
+    const roomId = _.get(router, 'query.roomId.0');
+
+    const [errorMessage, setErrorMessage] = useState(false);
+    const [serviceSpaceId, setServiceSpaceId] = useState();
+    const [removingLink, setRemovingLink] = useState(false);
+    const [serverSketches, setServerSketches] = useState({});
+    const [content, setContent] = useState(matrix.roomContents.get(roomId));
+    const [syncingServerSketches, setSyncingServerSketches] = useState(false);
+    const [isSpacedeckServerDown, setIsSpacedeckServerDown] = useState(false);
+    const path = getConfig().publicRuntimeConfig.authProviders.spacedeck.path?.replace(/[<>\s/:]/g, '') || 'spacedeck';
+
+    const spacedeck = auth.getAuthenticationProvider('spacedeck');
+
+    // Whenever the roomId changes (e.g. after a new sketch was created), automatically focus that element.
+    // This makes the sidebar scroll to the element if it is outside of the current viewport.
+    const selectedPadRef = useRef(null);
+    useEffect(() => {
+        selectedPadRef.current?.focus();
+    }, [roomId]);
+
+    useEffect(() => {
+        let cancelled = false;
+
+        const startLookingForFolders = async () => {
+            if (matrix.initialSyncDone) {
+                try {
+                    setServiceSpaceId(matrix.serviceSpaces.spacedeck);
```

Comment:
Correct me if I am wrong, but I don't think we need to store this service space ID in a separate state.
Check out what I did in https://github.com/medienhaus/medienhaus-spaces/commit/7258c4e409894c3d7cce879d0c6dc5d64dd02e3a -- I think we can do the same here... Let me know if it doesn't work or if you have any questions.